source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -84,10 +84,10 @@ void Cluster::onDnsHostAddOrUpdate(
 
   // Create an override transport socket options that automatically provides both SNI as well as
   // SAN verification for the resolved host if the cluster has been configured with TLS.
-  // TODO(mattklein123): If the host is an IP address we should not set SNI.
   Network::TransportSocketOptionsSharedPtr transport_socket_options =
       std::make_shared<Network::TransportSocketOptionsImpl>(
-          host_info->resolvedHost(), std::vector<std::string>{host_info->resolvedHost()});
+          !host_info->isIpAddress() ? host_info->resolvedHost() : "",
+          std::vector<std::string>{host_info->resolvedHost()});
```

Comment:
Will this only avoid SNI for the new forward proxy code then?   Do we do SNI/SAN anywhere else?  Grepping it looks like we only set TransportSocketOptions for the tcp proxy code, which surprises me.  But basically I'm wondering we should always pass hostname in and skip using it in TLS so it's reusable.

Reply:
The hard part there would be then having the "is this an IP address" logic embedded deep in the TLS code. But, could also go either way, will wait to see what @lizan and @PiotrSikora think.

Reply:
@alyssawilk `TransportSocketOptions` is only for overriding SNI/SAN (so they can be set from filters), we don't warn about the configuration error if tls_context has IP address SNI configured.

Reply:
@alyssawilk SNI is normally configured using [sni](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/auth/cert.proto#envoy-api-msg-auth-upstreamtlscontext) field in `tls_context`, and it doesn't get any automatic value. Same for SAN verification.